src/averageAges.js

```diff
@@ -37,7 +55,22 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const onlyWomen = people.filter(person => person.sex === 'f');
+  const womenNoChild = onlyWomen.map(woman => woman.died - woman.born);
+  const peopleWithMothers = people.filter(person => person.mother !== null);
+  const mothersNames = peopleWithMothers.map(person => person.mother);
+  const womenWithChild = onlyWomen
+    .filter(woman => mothersNames.includes(woman.name))
+    .map(woman => woman.died - woman.born);
+
+  let avrAge;
+
+  withChildren === undefined ? avrAge = womenNoChild.reduce((sum, age) =>
+    sum + age) / womenNoChild.length : avrAge
+    = womenWithChild.reduce((sum, age) =>
+      sum + age) / womenWithChild.length;
```

Comment:
same here, as with previous function, create a variable for needed array, don't repeat yourself